desktop/cmp/panel/impl/PanelHeader.ts

```diff
@@ -41,57 +62,57 @@ export const panelHeader = hoistCmp.factory({
             sideCls = `xh-panel-header--${side}`,
             compactCls = compact ? 'xh-panel-header--compact' : null;
 
-        // 1) Classic "top" title bar
-        if (!collapsed || vertical || isModal) {
-            return hbox({
-                className: classNames(className, compactCls),
+        // Return null if nothing to display.
+        if (isNil(displayedTitle) && isNil(displayedIcon) && isEmpty(displayedHeaderItems)) {
+            return null;
+        }
+
+        // Return a vertically oriented header if collapsed to left or right side.
+        if (isCollapsedToLeftOrRight) {
+            return vbox({
+                className: classNames(className, sideCls, compactCls),
+                flex: 1,
                 items: [
-                    icon || null,
+                    displayedIcon,
                     displayedTitle
                         ? box({
                               className: titleCls,
                               flex: 1,
                               item: span({className: `${titleCls}__inner`, item: displayedTitle})
                           })
                         : filler(),
-                    hbox({
-                        className: 'xh-panel-header__items',
-                        items: [
-                            ...(!collapsed || isModal ? headerItems : []),
-                            modalButton({panelModel}),
-                            collapseButton({panelModel})
-                        ],
-                        onDoubleClick: e => e.stopPropagation()
-                    })
+                    ...displayedHeaderItems
```

Comment:
In the vertical case we're not wrapping with ` className: 'xh-panel-header__items'` like we are below.

We could add for consistency, but it looked like we'd then need to add more specific styles to effectively undo the effect of the styles applied by the class, which are targeted at the horizontal orientation.  Wasn't sure that the inconsistency would throw off anything else, and in this case the only two header items that are possible are the two toggles.

Reply:
just a thought -- would it be helpful to just have a seperate component for the vertical collapsed case?